### PR TITLE
Sprint 2/issue 2#register user in database

### LINKED ISSUE #5: Register a user in database
### Tasks
- [x] Register with google account.
- [x] Register user with fist name, email and password.

### PR DESCRIPTION
**Pull request by Raibrr**
# Description

- Sign-in implementation with Google.
- Sign-up implementation with Firebase Auth.
- Save the user's data in Firestore.

## Screenshots
-Return a message if sign in with Google.
image

- Return a message if sign up with Firebase Auth
image

### CODE REVIEW

**Review by montsegv-2 — APPROVED** [review]

**Review by FranciscoSenestrari — APPROVED** [review]

**Review by Lizzflores — APPROVED** [review]
LGTM:D

File: FlightReservations/babel.config.js [hunk 1]
```diff
@@ -1,3 +1,9 @@
 module.exports = {
   presets: ['module:metro-react-native-babel-preset'],
+  plugins: [
+		["module:react-native-dotenv", {
+			"moduleName": "@env",
+			"path": ".env"
+			}]
+	]
```
Comment (YilKanda) [inline FlightReservations/babel.config.js:8]: Check the indentation of your file

File: FlightReservations/src/auth/SignInWithGoogle.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+			} else if (error.code === statusCodes.IN_PROGRESS) {
+				Alert.alert('operation (sign in) is in progress already: ');
+			} else if (error.code === statusCodes.PLAY_SERVICES_NOT_AVAILABLE) {
+				Alert.alert('play services not available or outdated: ');
+			} else {
+				Alert.alert('Error', error.toString());
+			}
+			console.log(error);
+	}
+}
```
Comment (YilKanda) [inline FlightReservations/src/auth/SignInWithGoogle.ts]: Check the indentation of your file

File: FlightReservations/src/auth/SignInWithGoogle.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+			if (error.code === statusCodes.SIGN_IN_CANCELLED) {
+				Alert.alert('Error', 'Login cancelled');
+			} else if (error.code === statusCodes.IN_PROGRESS) {
+				Alert.alert('operation (sign in) is in progress already: ');
+			} else if (error.code === statusCodes.PLAY_SERVICES_NOT_AVAILABLE) {
+				Alert.alert('play services not available or outdated: ');
+			} else {
+				Alert.alert('Error', error.toString());
+			}
+			console.log(error);
```
Comment (YilKanda) [inline FlightReservations/src/auth/SignInWithGoogle.ts]: Don't leave debugging code like console.log() on GitHub.

File: FlightReservations/src/auth/SignUp.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+		const addUser = await addUserInfo(user, name)
+	} catch (error) {
+		if (error.code === 'auth/email-already-in-use') {
+			Alert.alert('The email address is already in use!');
+		}
+
+		if (error.code === 'auth/invalid-email') {
+			Alert.alert('The email address is invalid!');
+		}
+	}
```
Comment (YilKanda) [inline FlightReservations/src/auth/SignUp.ts]: Check the indentation of your file.

Add a blank line.

File: FlightReservations/src/db/AddUserInfo.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+      firstName: displayName,
+      email: email,
+      uid: uid,
+      type: '',
+    })
+    .then(() => {
+      Alert.alert('User added succesfully');
+    })
+    .catch(error => console.log(error));
+};
```
Comment (YilKanda) [inline FlightReservations/src/db/AddUserInfo.ts]: Add a blank line.

File: FlightReservations/src/db/AddUserInfo.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    .set({
+      firstName: displayName,
+      email: email,
+      uid: uid,
+      type: '',
+    })
+    .then(() => {
+      Alert.alert('User added succesfully');
+    })
+    .catch(error => console.log(error));
```
Comment (YilKanda) [inline FlightReservations/src/db/AddUserInfo.ts:22]: Don't leave debugging code like console.log() on GitHub.

File: FlightReservations/types/env.d.ts [hunk 7]
```diff
@@ -0,0 +1,3 @@
+declare module '@env' {
+  export const API_URL: string;
+}
```
Comment (YilKanda) [inline FlightReservations/types/env.d.ts:3]: Add a blank line

File: FlightReservations/tsconfig.json [hunk 8]
```diff
@@ -6,5 +6,6 @@
 
     /* Completeness */
     "skipLibCheck": true                                 /* Skip type checking all .d.ts files. */
-  }
+  },
+  "typeRoots": ["./types"]
 }
```
Comment (YilKanda) [inline FlightReservations/tsconfig.json:3]: Add a blank line

File: FlightReservations/src/auth/SignInWithGoogle.ts [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+      .doc(user.uid)
+      .get()
+      .then(documentSnapshot => {
+        if (documentSnapshot.exists) {
+          Alert.alert('Signin')
+        } else {
+          addUserInfo(user);
+        }
+      })
+      .catch(error => console.log('error de usuario existe', error));
```
Comment (YilKanda) [inline FlightReservations/src/auth/SignInWithGoogle.ts]: Don't leave debugging code like console.log() on GitHub.

File: FlightReservations/.env.example [hunk 10]
```diff
@@ -0,0 +1,2 @@
+#Don't use '' to define the value
+API_URL=
```
Comment (YilKanda) [inline FlightReservations/.env.example:2]: Add a blank line

**Review by YilKanda — CHANGES_REQUESTED** [review]

**Review by YilKanda — APPROVED** [review]

**Pull request merged**